docs/dynamic-programming/cutting-a-rod.mdx

```diff
@@ -0,0 +1,178 @@
+---
+title: Cutting a rod
+slug: /dynamic-programming/cutting-a-rod
+---
+
+## Introduction
+
+Given a rod of length `n` and an array of prices of all pieces of length `1` to `n`, we need to determine the maximum price that can be obtained by cutting up the rod and selling its pieces.
+
+## Explanation
+
+The given rod of length `n` can be cut into sub-rods in various ways, each way resulting in a different selling price. We need to maximize the selling price.
+
+Let us have one example: 
+```
+Given, length of the rod n = 5, and price of rods of various lengths,
+
+Length - 1 | 2 | 3 | 4 | 5
+Price  - 2 | 5 | 7 | 8 | 10
+
+Ways of cutting the rod into sub-rods:
+
+1) Length 1 + Length 4
+
+  - Selling price = 2 + 8 = 10
+
+2) Length 2 + Length 3
+
+  - Selling price = 5 + 7 = 12
+
+3) Length 5
+
+  - Selling price = 10
+
+Here, we see that a maximum selling price of 12 is generated when the given rod is cut into
+sub-rods of length 2 and 3.
+```
+
+## Algorithm
+
+- Initialize an array `dp[]` such that the `ith` element stores the maximum price obtained by a rod of length `i`.
+
+```
+for i = 1 to n
+    current_max = i
+    for j = 0 to i - 1
+        current_max = max(cost[j] + dp[ i- j - 1], current_max)
+    dp[i] = current_max
+return dp[n]
+```
```

Comment:
Don't substitute algorithm steps with pseudocode. There is a reason why we have actual code implementations here. Use Algorithm to strictly describe how your code works step-wise